implementations/rust/ockam/ockam_abac/src/storage/policy_repository_sql.rs

```diff
@@ -112,9 +129,14 @@ impl PolicyRow {
     pub(crate) fn expression(&self) -> Result<Expr> {
         Ok(Expr::try_from(self.expression.as_str())?)
     }
+}
+
+impl TryFrom<PolicyRow> for Policy {
+    type Error = ockam_core::Error;
 
-    pub(crate) fn policy(&self) -> Result<Policy> {
-        Ok(Policy::new(self.expression()?))
+    fn try_from(row: PolicyRow) -> Result<Self, Self::Error> {
+        let resource = GlobalResource::strip_prefix(&row.resource).into();
```

Comment:
Could we use an extra field instead of the prefix? That would:
1. Allow to use it in queries
2. Avoid ambiguity/security issues with concatenating such sensitive strings

Reply:
The `prefix` thing is no longer necessary. It now does a hard check of the resource name, meaning that a resource named "tcp-outlet" is always going to be parsed as a global resource name.